README.md

```diff
@@ -53,6 +53,40 @@ There are some mandatory fields, each marked with "MANDATORY" in a comment.
 
 See [monitoring.md](docs/monitoring.md).
 
+### Attribute redaction
+
+While Kubernetes `v1.Secrets` are designed to separate secret fields into their
+own types, referenced in other types' fields such as container environment
+variable `secretKeyRefs`, some users may wish to prevent certain fields on
+certain types from being sent to Snyk, while still scanning other fields on that
+type.
+
+The scanner supports this sort of redaction, as an optional array of paths
+alongside group-version-kind scan config:
+
+```yaml
+- apiGroups: ["apps"]
+    versions: ["*"]
+    resources:
+      - replicasets
+      - daemonsets
+      - deployments
+      - statefulsets
+    redactions:
```

Comment:
I've gone back and forth about whether the "redaction" name fits our approach here, or if it implies we are only obscuring a key's value, but leaving the key (and structure) in place so that the system still understands how a resource is configured. It seems like the term could encompasses both ideas, just strictly looking at the dictionary definition.

I propose a more explicit `attributeRedactions` or `configRedactions` naming so that it's clear that we aren't excluding entire resources here (only at most large swaths of the config themselves).

Opening this thread to collect opinions.

Reply:
> or if it implies

tbh, as a non-native speaker, I had to check the dictionary what `redaction` means 😄 
Probably `attributeFilters` is easier to digest 

> but leaving the key (and structure) in place

This could get complex by time, imagine this manifest and our filter/reduction was `a.b`:
```json
{
    "a": {
        "b": [{
                "c": {"d": 1}
            },
            {
                "c": {"d": 2}
            }
        ],
        "e": "3"
    }
}
```
and if we are erasing the fields that match the filter, then we would have
```json
{
    "a": {
        "e": "3"
    }
}
```
but if we are redacting those fields, what are we going to return for children objects of the redacted field? And how useful would that be?
```json
{
    "a": {
       "b": ?????
        "e": "3"
    }
}
```

Reply:
@mmols-snyk `attributeFilters` sounds good to me.

@srlk we are deleting the whole subtree addressed by that field (as in your first example), so "redaction" is arguably a bad name anyway!

Reply:
Maybe we should go for `attributeRemovals` now that I look again, "filters" almost sounds like we're selecting based on its presence...

Reply:
Changed it to this in another commit, will squash when we're all happy.

I've kept the word "redact" in a few instances in comments and variable names where it actually does make sense: we _redact_ an object by removing some of its attributes, but we _remove_ the attributes themselves.

Let me know what you think.